docs/protocol/on-chain/channel-lifecycle.mdx

```diff
@@ -0,0 +1,375 @@
+- Maintain a record of the latest valid state
+- Follow <Tooltip content="A smart contract that validates state transitions according to application-specific rules.">adjudicator</Tooltip>-specific rules for signature requirements
+- Use application-specific data in the `state.data` field
+
+Each new state MUST:
+- Contain updated allocations reflecting the new distribution of assets
+- Be signed by the necessary participants according to application rules
+- Comply with the validation rules of the channel's adjudicator
+
+The on-chain contract remains unchanged during the active phase unless participants choose to checkpoint a state.
+
+:::tip Off-Chain Efficiency
+During the active phase, state updates occur entirely off-chain with zero gas costs and sub-second latency.
+:::
+
+## Checkpointing
+
+**Purpose**: Record a state on-chain without entering dispute mode.
+
+**Process**:
+
+1. Any participant MAY:
+   - Call the `checkpoint` function with a valid state and required proofs
+
+2. The contract MUST:
+   - Verify the submitted state via the adjudicator
+   - If valid and more recent than any previously checkpointed state, store it
+   - Emit a `Checkpointed` event with channelId
+
+**Benefits**:
+- Creates an on-chain record of current state
+- Shortens the effective history that could be challenged
+- Does not start a challenge period
+- Channel remains in `ACTIVE` status
+
+```mermaid
+graph LR
+    A[Active Channel] -->|checkpoint| B[Verify State]
+    B -->|Valid| C[Store State]
+    C --> D[Emit Event]
+    D --> E[Remain Active]
+    
+    style A fill:#90EE90
+    style E fill:#90EE90
+```
+
+:::note Optional Operation
+Checkpointing is optional but recommended for long-lived channels or after significant value transfers.
+:::
+
+## Closure - Cooperative
+
+**Purpose**: Close <Tooltip content="A secure communication pathway between participants that locks funds in an on-chain smart contract while enabling off-chain state updates.">channel</Tooltip> when all <Tooltip content="An entity (identified by a wallet address) that is part of a channel.">participants</Tooltip> agree on final <Tooltip content="A snapshot of the channel at a point in time, including fund allocations and application-specific data.">state</Tooltip>.
+
+**Process**:
+
+1. Any participant MAY:
+   - Prepare a final State where `state.data` contains magic number `CHANCLOSE` (7879)
+   - Collect signatures from all participants on this final state
+   - Call the `close` function with <Tooltip content="A unique identifier for a channel, computed as the hash of the channel configuration.">channelId</Tooltip>, final state, and any required proofs
+
+2. The contract MUST:
+   - Verify all participant signatures on the closing <Tooltip content="A cryptographic hash of a channel state, used for signature verification.">stateHash</Tooltip>
+   - Verify the state contains the `CHANCLOSE` magic number
```

Comment:
Update here and in the diagram below:

```suggestion
   - Verify all participant signatures on the closing <Tooltip content="A specific encoding of a State, used for signing and signature verification.">packedState</Tooltip>
   - Verify the state contains the `CHANCLOSE` magic number
```

Reply:
[f889dd9d34e1744c4d0d2598c3c590f87acb676b](https://github.com/layer-3/docs/pull/69/commits/f889dd9d34e1744c4d0d2598c3c590f87acb676b)